core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -56,7 +56,7 @@
     private String path;
     private boolean disablePathParams;
 
-    HttpRequestBuilder() {}
+    protected HttpRequestBuilder() {}
```

Comment:
Would it be possible to make this constructor package-local so that this class can't be extended by a user?

Reply:
Yes, but then `WebClientRequestBuilder` has to be in `common`. Or I can add an abstract builder and make `WebClient/HttpRequestBuilder` extend it and override return types but that's quite many boilerplate code 🤔.

Reply:
Yes, but shall we move `WebClientRequestBuilder` to `common`?

Reply:
We shouldn't move it to `common`. How about:

- Introducing public `AbstractHttpRequestBuilder` with a protected `buildRequest()` method and without `build()` method;
- Adding public final `HttpRequestBuilder extends AbstractHttpRequestBuilder`; and
- Adding public final `WebClientHttpRequestBuilder extends AbstractHttpRequestBuilder` ?

Reply:
I'm also not sure if the name `WebClientHttpRequestBuilder` is correct, because its eventual purpose is to execute a request and building it is just a part of the process. Should we rename it to something else? 🤔

Reply:
I like `WebClientHttpRequestPipeline` because to me it looks like a pipeline 😄

Reply:
How about `WebClientPreparation`? 🤔